coralproject/talk
# [CORL-2573]: Mark comments as read on scroll up out of view and at bottom

## What does this PR do?

These changes mark comments as seen when they are scrolled up and off the screen out of view in both the comment stream and the permalink view. They use the intersection observer to do this. These changes are for https://vmproduct.atlassian.net/browse/CORL-2573.

## What changes to the GraphQL/Database Schema does this PR introduce?

To the local schema, it adds `bottomOfCommentsInView` to know when the bottom has been reached and therefore all comments in view should be marked as read/seen.

## Does this PR introduce any new environment variables or feature flags? 

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can go to the stream and scroll through comments. See that comments you have scrolled up and off the screen are marked as seen. If you scroll a comment down off the screen, it will not be marked as read. When you get to the bottom of the stream, check that any comments in view are marked as seen. 

Check for the same behavior in the permalink view (Click `Share` on a comment and copy the link to get there). Choose a comment that has many replies to test that comments are also marked as seen when scrolled up and off the screen and when the bottom of the comments is reached, same as in the stream.
 
## How do we deploy this PR?

They will be deployed as part of the virtualized rendering changes, after the story tree backend updates are deployed.
